lib/std/Build.zig

```diff
@@ -554,7 +563,54 @@ fn determineAndApplyInstallPrefix(b: *Build) !void {
     b.resolveInstallPrefix(install_prefix, .{});
 }
 
+pub fn initCompdb(self: *Build) !void {
+    if (!self.enable_compdb) return;
+
+    self.compdb_entries_dir = self.makeTempPath();
+
+    const entry_paths_set = try self.allocator.create(StringHashMap(void));
+    entry_paths_set.* = StringHashMap(void).init(self.allocator);
+    self.compdb_entry_paths_set = entry_paths_set;
+}
+
+pub fn generateCompdb(self: *std.Build) !void {
+    if (!self.enable_compdb) return;
+
+    var entry_strings = ArrayList(u8).init(self.allocator);
+    defer entry_strings.deinit();
+
+    var entry_paths_set_key_it = self.compdb_entry_paths_set.?.keyIterator();
+    while (entry_paths_set_key_it.next()) |entry_path| {
+        const entry_path_abs = fs.cwd().realpathAlloc(self.allocator, entry_path.*) catch continue;
+
+        const entry_file = fs.openFileAbsolute(entry_path_abs, .{}) catch continue;
+        defer entry_file.close();
+
+        const entry_file_stat = entry_file.stat() catch continue;
+
+        var entry_file_contents: []const u8 = entry_file.readToEndAlloc(self.allocator, entry_file_stat.size) catch continue;
```

Comment:
`std.fs.readFileAlloc` does the trick here, `max_size` parameter is for prevening accidently OutOfMemory attacks by large files. Pass sometuing like 50MB here or similar (those json files can get huge)

Reply:
> Pass sometuing like 50MB here or similar (those json files can get huge)
The json file for chromium is 300mb (if i recall correctly, possibly even larger), so this needs to be larger to handle real projects

Reply:
> `std.fs.readFileAlloc` does the trick here, `max_size` parameter is for prevening accidently OutOfMemory attacks by large files. Pass sometuing like 50MB here or similar (those json files can get huge)

Done. I set it to a max of 8MiB. Remember, this is per entry. 8MiB should be more than enough for a single entry in any project I'm aware of.